pkgs/development/compilers/rust/rustc.nix

```diff
@@ -204,7 +206,9 @@ in stdenv.mkDerivation rec {
   passthru = {
     llvm = llvmShared;
     inherit llvmPackages;
-    tests = { inherit fd; } // lib.optionalAttrs stdenv.hostPlatform.isLinux { inherit firefox thunderbird; };
+    tests = {
+      inherit fd ripgrep wezterm;
```

Comment:
Admittedly not sure how I feel about adding ripgrep here, it's in a similar vein to fd after all. I suppose it can't hurt though, it'll get built either way after all.

Also, more importantly: OfBorg has never actually ran these tests (it's always timed out on all systems), so is there even a point to this?

Reply:
@zowoq You added the `passthru.tests` relatively recently in 9c26b06a5db9b47b7bbede202873b26effb668ce, what do you think? Do you build all of these when testing a bump? Because OfBorg certainly doesn't 😄

Reply:
> OfBorg has never actually ran these tests. It times out with firefox and thunderbird but usually the information is still useful

for example in https://github.com/NixOS/nixpkgs/pull/215530, IIUC this log means thunderbird and firefox are the only two derivations failed in `rustc.passthru.tests`

> error: build of '/nix/store/1zgrh3nwhjs4v2ssn8rzlcny08f177aj-firefox-109.0.1.drv', '/nix/store/vdvr7x4drs1p46h8b9hkn1kirmgzmwyk-thunderbird-102.7.2.drv' failed

Reply:
Ah, I did not see that, sorry. Maybe we remove just those, then?

Reply:
We usually test firefox and thunderbird when we upgrade rustc as both are channel blocker for nixos-unstable. Not sure if they need to be for that in `rustc.passthru.tests` though.

Reply:
Firefox at least is a channel blocker: https://github.com/NixOS/nixpkgs/blob/01a88ede6a2b64bd15b3b0fdcee30afb0b237edb/pkgs/top-level/release.nix#L111

Reply:
> Do you build all of these when testing a bump? 

Yes, I try to.

> Because OfBorg certainly doesn't

Didn't add it for ofborg, it's a convenience so people can `nix-build -A rustc.tests` instead of reading (or not reading) this:

https://github.com/NixOS/nixpkgs/blob/2590b99640bda4c1e84da440b56031a9f48d8578/pkgs/development/compilers/rust/1_67.nix#L3-L4

https://github.com/NixOS/nixpkgs/blob/2590b99640bda4c1e84da440b56031a9f48d8578/pkgs/development/compilers/rust/1_67.nix#L10

Reply:
For python packages and key libraries we are doing the same, so that people have a chance to know if updates break things and in my experience it has a very positive effect on updates for core packages especially in the python ecosystem.

Also I don't think it matters if ofborg can build those packages on a typical pull request if we know that a breakage in them will affect lots of people and block the channel.
If we find better packages for fd and ripgrep we can of course replace them.

I don't see anything actionable or controversial in this thread, just some questions and since this affects more people than just me, I would like to continue to merge this.

Reply:
> Also I don't think it matters if ofborg can build those packages on a typical pull request if we know that a breakage in them will affect lots of people and block the channel.

I don't see what you mean by this, can you elaborate/rephrase?

Reply:
I wanted to say that we don't care if ofborg fails to build firefox/thunderbird if that is a channel blocker.